core/src/avm1/globals/text_field.rs

```diff
@@ -127,20 +129,20 @@ fn get_new_text_format<'gc>(
     activation: &mut Activation<'_, 'gc, '_>,
     _args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    let tf = text_field.new_text_format();
-    Ok(TextFormatObject::new(activation, tf).into())
+    let text_format = text_field.new_text_format();
+    Ok(TextFormatObject::new(activation, text_format).into())
 }
 
 fn set_new_text_format<'gc>(
     text_field: EditText<'gc>,
     activation: &mut Activation<'_, 'gc, '_>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    let tf = args.get(0).unwrap_or(&Value::Undefined);
-
-    if let Value::Object(tf) = tf {
-        if let Some(tf) = tf.as_text_format_object() {
-            text_field.set_new_text_format(tf.text_format().clone(), &mut activation.context);
+    if let [Value::Object(object), ..] = args {
```

Comment:
Sadly, I couldn't find a way to do with less indentation. I tried to chain the last two `if`s to a single `if let Some(NativeObject::TextFormat(tf))`, but encountered Rust borrow checker errors. Any ideas are welcome.